lib/evaluate/check-expression.cc

```diff
@@ -277,8 +277,7 @@ class IsSimplyContiguousHelper
   }
 
   Result operator()(const ArrayRef &x) const {
-    return (x.base().IsSymbol() || x.base().Rank() == 0) &&
-        CheckSubscripts(x.subscript()) && (*this)(x.base());
+    return CheckSubscripts(x.subscript()) && (*this)(x.base());
```

Comment:
The reason for the current predicate is to ensure that `A(:)%B(1)` is not a false positive for simple contiguity.  It might be broken, but we need to ensure that this fix keeps that case negative.